tests/testing.ts

```diff
@@ -275,6 +283,65 @@ export function runTests(force: boolean, onlyResults: boolean, unsafe = false, e
   }
 }
 
+function isRustInstalled() {
+  try {
+    execSync('rustc --version', { stdio: 'inherit' });
+    execSync('cargo --version', { stdio: 'inherit' });
+    execSync('rustup --version', { stdio: 'inherit' });
+  } catch (e) {
+    return false;
+  }
+  return true;
+}
+
+function cairo1setup(): boolean {
+  // check if rust is installed
+  if (!isRustInstalled()) {
+    console.log(
+      "Please make sure you've installed rust alongside with rustup and cargo from here: https://www.rust-lang.org/tools/install",
+    );
+    return false;
+  }
+
+  // check if cairo1 directory exists and setup has been done
+  if (
+    !fs.existsSync(CAIRO1_DIR) ||
+    !fs.existsSync(CAIRO1_CARGO_TOML) ||
+    !fs.existsSync(CAIRO1_COMPILE_BIN) ||
+    !fs.existsSync(SIERRA_RUN_BIN)
+  ) {
+    console.log(
+      "Please make sure you've done the cairo1 setup from here: https://github.com/starkware-libs/cairo#readme",
+    );
+    return false;
+  }
+  return true;
+}
+
+function doCairo1Setup() {
+  if (!isRustInstalled) {
+    throw new Error(
+      "Please make sure you've installed rust alongside with rustup and cargo from here: https://www.rust-lang.org/tools/install",
+    );
+  }
+  try {
+    console.log('Cloning cairo1 repo...');
+    execSync('git clone https://github.com/starkware-libs/cairo', { stdio: 'inherit' });
+  } catch (e) {
+    if (e instanceof Error) {
+      throw `Failed to clone the cairo1 repo: ${e.message}`;
+    }
+    throw e;
+  }
+  // follow cairo1 README setup instructions
+  try {
+    console.log('Installing cairo1 dependencies...');
+    execSync(`cargo test --manifest-path ${CAIRO1_CARGO_TOML}`, { stdio: 'inherit' });
+  } catch (e) {
+    throw new Error('Failed to compile cairo1 and its dependencies');
+  }
+}
```

Comment:
Can't we use binaries from their github? https://github.com/starkware-libs/cairo/releases

Reply:
Binaries are specific to linux machines architecture, not sure whether it works for other machines

Reply:
True